packages/errors/lib/data-store-error.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @module @dotcom-reliability-kit/errors/lib/data-store-error
+ */
+
+const OperationalError = require('./operational-error');
+
+/**
+ * Class representing an error in an application's data store.
+ */
+class DataStoreError extends OperationalError {
+	/**
+	 * @readonly
+	 * @access public
+	 * @type {string}
+	 */
+	name = 'DataStoreError';
```

Comment:
**question:** I'm still getting my head around JSDocs. In this code block it looks like `name`  is not commented out but neither is it valid code as it is not preceded with `var`, `let`, or`const`.

I'm also unsure in this context what `name` refers to as it does not seem to be referenced elsewhere in this file.

Can you help me understand pls?

Reply:
Ooh this is a [public class field](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields), it's a relatively new (Node.js 14+) feature of JavaScript which allows you to define instance properties on the class definition rather than having to set `this.whatever` in the constructor.

So in this `DataStoreError` class it means that:

```js
const error = new DataStoreError('Something went wrong');
error.name === 'DataStoreError'; // true
```

Even more exciting than this is the existence of [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) (not used here), which are only accessible by methods on the instance of a class so we have actual private properties in classes now.

Reply:
These are _kind_ of equivalent (with a few minor differences explained in the Mozilla docs):

```js
// Class fields
class Example {
    myProperty = 123;
}
```

```js
// Constructor
class Example {
    constructor() {
        this.myProperty = 123;
    }
}
```

Reply:
Amazing - thanks for the explanation 🙇